registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/BioSdkUtil.java

```diff
@@ -0,0 +1,66 @@
+package io.mosip.registration.processor.packet.storage.utils;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import io.mosip.kernel.biometrics.commons.CbeffValidator;
+import io.mosip.kernel.biometrics.constant.BiometricFunction;
+import io.mosip.kernel.biometrics.constant.BiometricType;
+import io.mosip.kernel.biometrics.entities.BIR;
+import io.mosip.kernel.biosdk.provider.factory.BioAPIFactory;
+import io.mosip.kernel.biosdk.provider.spi.iBioProviderApi;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.kernel.core.util.CryptoUtil;
+import io.mosip.registration.processor.core.constant.MappingJsonConstants;
+import io.mosip.registration.processor.core.idrepo.dto.Documents;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+
+@Component
+public class BioSdkUtil {
+
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(BioSdkUtil.class);
+
+	/** The utilities. */
+	@Autowired
+	Utilities utilities;
+
+	/** The bio api factory. */
+	@Autowired(required = false)
+	private BioAPIFactory bioApiFactory;
+
+	/** modality arrays that needs to be tagged */
+	@Value("#{'${mosip.regproc.quality.classifier.tagging.quality.modalities}'.split(',')}")
+	private List<String> modalities;
+
+	public boolean authenticateBiometrics(String uin, String individualType,
+			List<io.mosip.kernel.biometrics.entities.BIR> list) throws Exception {
+		regProcLogger.info("BioSdkUtil :: authenticateBiometrics :: Fetching info from id repo based on uin ");
+		boolean status = false;
+		List<Documents> docs = utilities.retrieveIdrepoDocument(uin);
+		String data = null;
+		for (int i = 0; i < docs.size(); i++) {
```

Comment:
This for loop is not required here in your case. We are not getting INDIVIDUAL_BIOMETRICS here. Instead we are getting introducer/supervisor/operator biometrics here.

Reply:
for particular UIN response will have list of documents  in that list need to fetch doc relevant to individual biometrics
so using for loop and break 
"documents" :[
{
    "category":"proofOfAddress",
	"value":"asdbuasgdjhkasdfgASDJBasd"
},
{
    "category":"proofOfIdentity",
	"value":"wuerhoqiwuehruqwerwerqwer"
},
{
    "category":"individualBiometrics",
	"value":"erhtwierheruiafbljkjblii"
}
]